rdkit_valgrind.suppressions

```diff
@@ -0,0 +1,3235 @@
+{
+   uninitialised value used in logic 4
+   Memcheck:Cond
+   fun:__mpn_addmul_1
+   obj:*
+   obj:*
+   fun:__mpn_mul
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   fun:__printf_fp
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+}
+{
+   uninitialised value used in logic 5
+   Memcheck:Cond
+   fun:__mpn_addmul_1
+   obj:*
+   obj:*
+   fun:__mpn_mul
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+}
+{
+   memory possibly lost 8
+   Memcheck:Leak
+   match-leak-kinds: possible
+   fun:_Znwm
+   fun:RDValue
+   fun:_ZN5RDKit4Dict6setValISt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS8_EEEEvRKS8_RT_
+   fun:setProp<unsigned int>
+   fun:_ZN5RDKit9Chirality18assignAtomCIPRanksERKNS_5ROMolERSt6vectorIjSaIjEE
+   fun:_ZN5RDKit9Chirality21assignAtomChiralCodesERNS_5ROMolERSt6vectorIjSaIjEEb
+   fun:_ZN5RDKit6MolOps21assignStereochemistryERNS_5ROMolEbbb
+   fun:_ZN5RDKit11SmilesToMolERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKNS_18SmilesParserParamsE
+   fun:_ZN5RDKit17SmilesMolSupplier11processLineENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE
+   fun:_ZN5RDKit17SmilesMolSupplier4nextEv
```

Comment:
Most of these look like the kind of leak that would happen if a test forgot to clean up a mol that it used.

Do you still have the raw valgrind logs? It might be good to create a RDKit issue and attach them.

Reply:
Yeah, there are many, many of these.
Is it possible to add suppressions at the "test" level (i.e. at the level of the executable that ctest runs)? Then we could "opt in" all the tests that currently fail and remove them from that list as we fix them.

Reply:
I don't have too much experience with valgrind, so I don't know if that's even possible.
Actually, I took the other way: I am going over the memory errors I have found, and I am fixing them as best as I can. Most of the tests are already done, only a few left (the ones that take longer to run). I will put up another PR when I'm done.

Reply:
I added a new PR with the fixes, please see #2123.

Reply:
I don't know if it is technically feasible to separate the leaks that are in the tests from the leaks that are in the production code.

For instance, if a test created a Mol and forgot to clean it up, that would look the same as if the test created a Mol and passed it to a function that was supposed to destroy it, but failed.